### PR TITLE
WIP: Elm 0.19 support

### PR DESCRIPTION
**Pull request by avh4**

### CODE REVIEW

**Comment by avh4** [comment]
Continued work on https://github.com/avh4/elm-upgrade/tree/dev-0.19-ci-blocked-on-core-elm, which can't be merged to a branch with CI until Elm 0.19 binaries are available.

File: upgrade.js [hunk 1]
```diff
@@ -90,27 +90,23 @@ function main (knownUpgrades) {
...
   )
 
-  var elmUsage = childProcess.execFileSync(elm)
-  var elmVersion = elmUsage.toString().split('\n')[0].split(' - ')[0].trim()
-  if (!elmVersion.match(/^Elm Platform 0\.18\./)) {
-    process.stderr.write('ERROR: Elm 0.18 required, but found ' + elmVersion + '\n' + howToInstallElm())
+  var elmUsage = childProcess.execFileSync(elm, [ '--version' ])
+  var elmVersion = elmUsage.toString().split('\n')[0]
+  if (!elmVersion.match(/^0\.19\./)) {
+    process.stderr.write('ERROR: Elm 0.19 required, but found ' + elmVersion + '\n' + howToInstallElm())
```
Comment (eeue56) [inline upgrade.js]: Could use template strings instead here

File: upgrade.js [hunk 2]
```diff
@@ -90,27 +90,23 @@ function main (knownUpgrades) {
...
-  process.stdout.write('INFO: Found ' + elmVersion + '\n')
+  process.stdout.write('INFO: Found elm ' + elmVersion + '\n')
 
   var elmFormat = localFindBinary('elm-format', 'ERROR: elm-format was not found on your PATH.  Make sure you have elm-format installed.\n' + howToInstallElmFormat())
 
   var elmFormatUsage = childProcess.execFileSync(elmFormat)
   var elmFormatVersion = elmFormatUsage.toString().split('\n')[0].trim().split(' ')[1]
-  if (semver.lt(elmFormatVersion, '0.5.2-alpha')) {
-    process.stderr.write('ERROR: elm-format >= 0.5.2-alpha required, but found ' + elmFormatVersion + '\n' + howToInstallElmFormat())
+  if (semver.lt(elmFormatVersion, '0.7.1-beta')) {
```
Comment (eeue56) [inline upgrade.js]: :thought_balloon: In theory we can abstract this out. If we had a build script that injected the version, then publishing a new version of elm-upgrade would be trivial

File: upgrade.js (hunk 1 above)
Comment (avh4) [reply upgrade.js]: I don't understand the difference between all the different versions of nodejs...  are template strings supported in all versions that people are likely to have?  Apparently the default nodejs version on Travis-CI is node 0.10, which didn't seem to support template strings.

File: upgrade.js (hunk 2 above)
Comment (avh4) [reply upgrade.js]: I think that will be worth thinking about for 0.20.  At the moment, it hasn't seemed like there's a ton of consistency between the 0.18 and 0.19 upgrades, so I suspect it's likely future Elm versions upgrades will still need a lot of manual attention to upgrade elm-upgrade.

**Comment by avh4** [comment]
(Also, note see https://github.com/avh4/elm-upgrade/pull/26#issuecomment-383321006 for continue work on 0.19)

File: upgrade.js (hunk 1 above)
Comment (eeue56) [reply upgrade.js]: This has some stats on what versions of node people are using: https://nodesource.com/node-by-numbers

Generally, people who use Elm also have modern versions of Node installed. I'm not sure if there's a good way to identify that. At least elm-test requires Node > 4.7: https://github.com/rtfeldman/node-test-runner/blob/master/package.json#L7

**Comment by avh4** [comment]
Closing this PR, since CI isn't going to work until Elm 0.19 has publicly released binaries.  Ongoing work has continued in the `rc` branch.

**Pull request closed without merging**